script/upgrade/Mode-100-to-100.s.sol

```diff
@@ -0,0 +1,351 @@
+        else if (isTestnet(_network)) return Multisig(0x14b1812260CB993bca69f204bC43586322d246d0);
+        else revert("Invalid network");
+    }
+
+    /////////////////////////////////////////////
+    // -------------- STATE VARIABLES ---------//
+    /////////////////////////////////////////////
+
+    GaugesDaoFactory factory;
+
+    GaugePluginSet modePluginSet;
+    GaugePluginSet bptPluginSet;
+
+    /// @dev Mode multisig executing via the dao
+    Multisig modeMultisig;
+
+    /// @dev Mode dao owning the contracts
+    DAO modeDAO;
+
+    Lock lockMode;
+    Lock lockBPT;
+    SimpleGaugeVoter voterMode;
+    SimpleGaugeVoter voterBPT;
+
+    address[] aragonSigners;
+    address[] modeSigners;
+
+    /////////////////////////////////////////////
+    // -------------- RUN FUNCTION ------------//
+    /////////////////////////////////////////////
+
+    function run() public {
+        bool isTestMode = false;
+        uint256 aragonProposalId = _actionUpgrade(isTestMode);
+        console.log("Aragon Proposal ID: ", aragonProposalId);
+    }
+
+    function _actionUpgrade(bool isTestMode) internal returns (uint aragonProposalId) {
+        setModeSigners();
+        setAragonSigners();
+        _retrieveDeployment(vm.envAddress("FACTORY_ADDRESS"));
+
+        _validateUpgrade();
+
+        _startBroadcastOrPrank(isTestMode);
+        {
+            IDAO.Action[] memory actions = buildActions();
+            aragonProposalId = _createAragonMsigProposal(actions);
+        }
+        _stopBroadcastOrPrank(isTestMode);
+    }
+
+    function _validateUpgrade() internal {
+        Options memory options;
+
+        string[] memory exclude = new string[](1);
+        // disable initializers is invoked but the custom unsafe allow option is not set in the natspec
+        exclude[0] = "lib/osx/packages/contracts/src/core/plugin/PluginUUPSUpgradeable.sol";
+        options.exclude = exclude;
+
+        options.referenceContract = "SimpleGaugeVoter.sol";
+        Upgrades.validateUpgrade("SimpleGaugeVoter_v1_1_0.sol:SimpleGaugeVoterV1_1_0", options);
+    }
+
+    function buildActions() internal returns (IDAO.Action[] memory) {
+        // action 1: deploy new impls
+        address voterImplNew = address(new SimpleGaugeVoterV1_1_0());
+
+        // action 2: upgradeTo
+        IDAO.Action[] memory actions = new IDAO.Action[](2);
+        actions[0] = IDAO.Action({
+            to: address(voterMode),
+            value: 0,
+            data: abi.encodeCall(voterMode.upgradeTo, (voterImplNew))
+        });
+
+        actions[1] = IDAO.Action({
+            to: address(voterBPT),
+            value: 0,
+            data: abi.encodeCall(voterBPT.upgradeTo, (voterImplNew))
+        });
+
+        return actions;
+    }
+
+    function _startBroadcastOrPrank(bool isTestMode) internal {
+        address signer = vm.envAddress("SIGNER_ADDRESS");
+        if (isTestMode) {
+            vm.startPrank(signer);
```

Comment:
Is there a situation where this signer won't be an Aragon signer? Could we use `aragonSigner[0]`?

Reply:
No because in the event of a broadcast you need access to the private key